rotkehlchen/chain/bitcoin/bch/utils.py

```diff
@@ -58,55 +23,40 @@ def is_valid_bitcoin_cash_address(address: str) -> bool:
         address = _PREFIX + ':' + address
     elif colon_count > 1:
         return False
-    prefix, base32string = address.split(':')
-    decoded = _b32decode(base32string)
-    return _polymod(_prefix_expand(prefix) + decoded) == 0
-
-
-def _b32decode(inputs: str) -> list:
-    return [_CHARSET.find(letter) for letter in inputs]
 
+    try:
+        BchBech32Decoder.Decode(hrp=_PREFIX, addr=address)
+    except ValueError:
+        return False
 
-def _address_type(address_type: str, version: str | int) -> tuple[str, int, bool]:
-    for mapping in _VERSION_MAP[address_type]:
-        if version in (mapping[0], mapping[1]):
-            return mapping
-    raise ValueError('Invalid Address')
+    return True
 
 
-def _code_list_to_string(code_list: list[int]) -> bytes:
-    output = b''
-    for code in code_list:
-        output += bytes([code])
-    return output
+def convert_version(version: int, target_type: Literal['legacy', 'cash']) -> int:
```

Comment:
hmmm all this logic here I think I got it from external code/library. Did you also do so here?